src/shout_subgroup/modify_subgroup.py

```diff
@@ -187,3 +186,99 @@ async def subgroup_handler(update: Update, context: ContextTypes.DEFAULT_TYPE) -
     except Exception:
         logging.exception("An unexpected exception occurred")
         await update.message.reply_text("Whoops 😅, something went wrong on our side.")
+
+
+async def remove_users_from_existing_subgroup(
+        db: Session,
+        telegram_chat_id: int,
+        subgroup_name: str,
+        usernames: set[str]
+) -> SubgroupModel:
+    if not await is_group_chat(telegram_chat_id):
+        msg = f"Can't kick members from subgroup because telegram chat id {telegram_chat_id} is not a group chat."
+        logging.info(msg)
+        raise NotGroupChatError(msg)
+
+    subgroup = await find_subgroup_by_telegram_group_chat_id_and_subgroup_name(db, telegram_chat_id, subgroup_name)
+    if not subgroup:
+        msg = f"Subgroup {subgroup_name} should exist for telegram group chat id {telegram_chat_id}."
+        logging.info(msg)
+        raise SubGroupDoesNotExistsError(msg)
+
+    # If we can't find all the users, then it means we have not saved them yet.
+    users_to_be_removed = await find_users_by_usernames(db, usernames)
+    if len(users_to_be_removed) != len(usernames):
+        raise UserDoesNotExistsError("All the usernames are not in the database.")
+
+    # Find all the users who aren't in the group, then add them
+    for user in users_to_be_removed:
+        # How Equivalency is Checked:
+        # __eq__ Method:
+        # SQLAlchemy model instances, like those created with the `UserModel`, have an `__eq__` method
+        # that checks equivalency based on the primary key by default.
+        # This means that two instances of `UserModel` are considered equal
+        # if their primary key (`user_id`) values are the same.
+        if user in subgroup.users:
+            subgroup.users.remove(user)
+
+    # Commit the transaction
+    db.commit()
+    db.refresh(subgroup)
+
+    return subgroup
+
+
+async def remove_subgroup_member_handler(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
+    """
+    Handles removing members from subgroups.
+    This function should not handle business logic,
+    or storing data. It will delegate that responsibility
+    to other functions. Similar to controllers from the MVC pattern.
+    :param update:
+    :param context:
+    :return:
+    """
+    args = context.args
+
+    # Quick guard clause
+    if len(args) < 2:
+        msg = "You didn't use this command correctly. Please type /kick <group_name> @alice @bob ... @zack"
+        await update.message.reply_text(msg)
+        return
+
+    subgroup_name = args[0]
+
+    usernames = {name.replace("@", "") for name in set(args[1:])}  # removing mention from args for usernames
+    if not await usernames_valid(usernames):
+        await update.message.reply_text("Not all the usernames are valid. Please re-check what you entered.")
+    try:
+
+        subgroup = await remove_users_from_existing_subgroup(session, update.effective_chat.id, subgroup_name, usernames)
```

Comment:
I didn't pull this into a `_handle_remove_subgroup_member` b/c I don't think the number of lines of logic needed to be abstracted